src/eg/src/ballot.rs

```diff
@@ -29,8 +37,11 @@ pub enum BallotState {
 /// An encrypted ballot.
 #[derive(Debug, Serialize, Deserialize)]
 pub struct BallotEncrypted {
+    /// The index of ballot style that this ballot belongs to.
+    pub ballot_style_index: BallotStyleIndex,
+
     /// Contests in this ballot
-    pub contests: Vec1<ContestEncrypted>,
+    pub contests: BTreeMap<ContestIndex, ContestEncrypted>,
```

Comment:
This is a design change worth thinking about carefully. My thinking was that the `contests` vec must have exactly the number of elements as indicated by the ballot style referred to by its `ballot_style_index`.
Allowing every ballot to specify its own set of contests introduces the possibility of a subtle inconsistency: a ballot could specify votes for a contest not present in its ballot style. Indeed, the `ContestNotInManifest` error condition was defined, as would be appropriate for validating a ballot style definition. But invalid ballots could produce `BallotStyleNotInManifest` and/or `ContestNotInBallotStyle` errors.